### PR TITLE
Remove legacy WebView browser

### PR DESCRIPTION
**Pull request by barijaona**

### CODE REVIEW

File: Vienna/Sources/Shared/HelperFunctions.m [hunk 1]
```diff
@@ -117,15 +117,15 @@ BOOL hasOSScriptsMenu(void)
...
+        // Clean up user-entered URLs that might contain umlauts,
         // diacritics and other IDNA related stuff in the domain, or whatever
         // may hide in filenames and arguments.
         NSPasteboard *pasteboard = [NSPasteboard pasteboardWithUniqueName];
         [pasteboard declareTypes:@[NSPasteboardTypeString] owner:nil];
         @try {
             if ([pasteboard setString:urlString
                               forType:NSPasteboardTypeString]) {
-                url = [WebView URLFromPasteboard:pasteboard];
+                url = [NSURL URLFromPasteboard:pasteboard];
```
[the remark this answers is not in the document]
Comment (barijaona) [reply Vienna/Sources/Shared/HelperFunctions.m:128]: NSURL.URLFromPasteBoard does not behave exactly like WebView.URLFromPasteBoard, but I have checked that it handles special characters, like those in http://ουτοπία.δπθ.gr/نجيب_محفوظ/ (used by `testURLIDNinArticleView()`)

**Comment by barijaona** [comment]
The test fails on macOS 12 but succeeds in macOS 13 ?

File: Vienna/Sources/Shared/HelperFunctions.m (hunk 1 above)
[the remark this answers is not in the document]
Comment (barijaona) [reply Vienna/Sources/Shared/HelperFunctions.m:128]: The test fails on macOS 12 but succeeds in macOS 13 ?

[the remark this answers is not in the document]
Comment (barijaona) [reply Vienna/Sources/Shared/HelperFunctions.m:128]: On macOS 13, conversion of the above mentioned string does not even require entering in this part of `urlFromUserString()`, as the conversion to NSURLComponents is able to do the trick…

[the remark this answers is not in the document]
Comment (barijaona) [reply Vienna/Sources/Shared/HelperFunctions.m:128]: I confirm that the problem is present in macOS 12 : `[NSURL URLFromPasteboard:pasteboard]` returns `nil` (as does `[NSURLComponents componentsWithString:urlString]`, contrary to macOS 13.

File: Vienna/Sources/Shared/HelperFunctions.m [hunk 2]
```diff
@@ -125,7 +126,11 @@ BOOL hasOSScriptsMenu(void)
         @try {
             if ([pasteboard setString:urlString
                               forType:NSPasteboardTypeString]) {
-                url = [NSURL URLFromPasteboard:pasteboard];
+                if (@available(macOS 13, *)) {
+                    url = [NSURL URLFromPasteboard:pasteboard];
+                } else {
+                    url = [WebView URLFromPasteboard:pasteboard];
+                }
```
[the remark this answers is not in the document]
Comment (barijaona) [reply Vienna/Sources/Shared/HelperFunctions.m]: Updated

**Comment by TAKeanice** [comment]
Yes, I wanted to all the time, but I hope to find a moment for it this weekend!

File: Vienna/Sources/Main window/ArticleCellView.m
[the remark this answers is not in the document]
Comment (barijaona) [reply Vienna/Sources/Main window/ArticleCellView.m:1]: I think the code can be safely removed… It was initially a transposition of existing code in ArticleView.m/BrowserPane.m, but it seems not relevant anymore:
- in Unified view, there is no possibility for user to initiate load cancellation
- we don't support WebKit plugins anymore

In fact, I am unable to identify circumstances triggering this delegate.

**Pull request merged**